Project1/Map.cpp

```diff
@@ -0,0 +1,135 @@
+#include "Map.h"
+Map::Map(int width, int height) : _grid(width, std::vector<GridCell>(height, GridCell())) {
+	insertAdjacencies();
+}
+GridCell & Map::getGridCell(int x_pos, int y_pos)
+{
+	return _grid[x_pos][y_pos];
+}
+void Map::insertAdjacencies() {
+	int width = _grid.size();
+	int height = _grid[0].size();
+	for (int x = 0; x < width; x++) {
+		for (int y = 0; y < height; y++) {
+			if (x > 0) {
+				_grid[x][y].addAdjacentCell(&(_grid[x - 1][y]), MobilityList({ 10, 10, 10 }), MobilityList({true, true, true}));
+			} if (x < width - 1) {
+				_grid[x][y].addAdjacentCell(&(_grid[x + 1][y]), MobilityList({ 10, 10, 10 }), MobilityList({ true, true, true }));
+			}if (y > 0) {
+				_grid[x][y].addAdjacentCell(&(_grid[x][y - 1]), MobilityList({ 10, 10, 10 }), MobilityList({ true, true, true }));
+			} if (y < height - 1) {
+				_grid[x][y].addAdjacentCell(&(_grid[x][y + 1]), MobilityList({ 10, 10, 10 }), MobilityList({ true, true, true }));
+			}
+		}
+	}
+}
+bool Map::moveUnit(GridCell* start, GridCell* destination) {
+	if (!start->getTile().hasUnit() || destination->getTile().hasUnit()) {
+		return false;
+	}
+	else {
+		_unit_to_cell.at(start->getTile().getUnit()) = destination; // Map the Unit to the destination.
+		destination->getTile().insertUnit(start->getTile().removeUnit()); // Put the Unit in the destination Tile
+		return true;
+	}
+}
+bool Map::insertUnit(Unit* new_unit, GridCell* destination) {
+	if (destination->getTile().hasUnit()) {
+		return false;
+	}
+	else {
+		_unit_to_cell.insert(std::pair<Unit*, GridCell*>(new_unit, destination));// Map the Unit to the destination.
+		destination->getTile().insertUnit(new_unit);
+		return true;
+	}
+}
+void Map::removeUnit(Unit* unit) { //throws exceptions if the space is not empty.
+	_unit_to_cell.at(unit)->getTile().removeUnit();
+	_unit_to_cell.erase(unit);
+}
+
+std::vector<GridCell*> Map::getAccesibleCells(Unit* unit) {
+	findShortestPaths(_unit_to_cell.at(unit), unit->getStats().getAttribs()[AttribType::values::MOV], unit->getStats().getMobilityType());
+	std::vector<GridCell*> cells = std::vector<GridCell*>();
+	for (std::map<GridCell*, prev_cost_pair>::iterator it = _shortest_path_map.begin(); it != _shortest_path_map.end(); it++) {
+		cells.push_back(it->first);
+	}
+	return cells;
+}
+CellPath Map::getShortestPath(GridCell* start, GridCell* destination) {
+	findShortestPaths(start);
+	if (start == destination) {
```

Comment:
Start has a zero length path to itself; you shouldn't need the special case here.